src/social_network_link_prediction/other_methods/information_theory/path_entropy.py

```diff
@@ -0,0 +1,45 @@
+import networkx as nx
+import math
+from scipy.sparse import lil_matrix
+
+def path_entropy(G:nx.Graph, max_path:int = 3) -> lil_matrix:
+
+    similarity_matrix = lil_matrix((G.number_of_nodes(), G.number_of_nodes()))
+
+    missing_edges = list(nx.complement(G).edges())
+
+    for elem in missing_edges:
+        paths = nx.all_simple_paths(G, elem[0], elem[1], max_path)
+        tmp = 0
+        for i in range(2,(max_path+1)):
+            tmp += (1 / (i - 1)) * simple_path_entropy(paths = paths, G = G)
+        tmp = tmp - new_link_entropy(G, elem[0], elem[1])
+        similarity_matrix[elem[0],elem[1]] = tmp
+    return similarity_matrix
```

Comment:
Deve ritornare un `csr_matrix`

```suggestion
def path_entropy(G:nx.Graph, max_path:int = 3) -> csr_matrix:

    similarity_matrix = lil_matrix((G.number_of_nodes(), G.number_of_nodes()))

    missing_edges = list(nx.complement(G).edges())

    for elem in missing_edges:
        paths = nx.all_simple_paths(G, elem[0], elem[1], max_path)
        tmp = 0
        for i in range(2,(max_path+1)):
            tmp += (1 / (i - 1)) * simple_path_entropy(paths = paths, G = G)
        tmp = tmp - new_link_entropy(G, elem[0], elem[1])
        similarity_matrix[elem[0],elem[1]] = tmp
    return similarity_matrix.tocsr()
```

Reply:
Risolto in 5b4ab4d